src/vm/runners/cairo_runner.zig

```diff
@@ -26,6 +26,76 @@ const expectEqual = std.testing.expectEqual;
 const expectError = std.testing.expectError;
 const expectEqualSlices = std.testing.expectEqualSlices;
 
+/// Tracks the step resources of a cairo execution run.
+const RunResources = struct {
```

Comment:
Shouldn't we add a `init` method here to init a `RunResources` instance from a number of steps:

```zig
pub fn consumed(n_steps: usize) Self {
    return .{.n_steps = n_steps};
}
```

wdt?